### PR TITLE
🚀 Support MultiGPU/SiggleGPU in one base_trainer, Refactor project.

### PR DESCRIPTION
**Pull request by dathudeptrai**
This PR include several changes:

- [x] Support Single/Multi-GPU in GanBasedTrainer and Seq2SeqBasedTrainer.
- [x] More abstract for BasedTrainer, now you only need override **compile, compute_loss, generate_intermediate** function for new models.
- [x] All dataloader return dictionary with the key name match exactly with model input name.
- [x] 20% training speed-up for Tacotron-2 with tensorflow 2.3.0rc1, change `BatchNormalization` to `SyncBatchNormalization`.
- [x] Remove `attention_mask` for all model since we can compute it based on `input_ids` inside a function call.
- [x] Change init `learning_rate` of `melgan.stft` and `multiband_melgan` to `0.0005`, clip value for `stft loss` to prevent expoded/nan loss.
- [x] Test training for `tacotron-2` with multi-gpu.
- [x] Test training `melgan/melgan.stft/mb_melgan` with multi-gpu.
- [x] Test `FastSpeech/FastSpeech2` with multi-gpu.
- [x] `CUDA_VISIBLE_DEVICES=0,1,2` to training with multi-gpu. 
- [x] Replace some typo on README.
- [x] `batch_size` in config is `batch_size` for each GPU (replica), for multi-GPU a `global_batch_size` = `batch_size` * `n_gpus.`
- [x] Test the code with Tensorflow 2.2.0 and Tensorflow 2.3.0rc1, Tensorflow_addons 0.9.1 and Tensorflow_addons 0.10.0.
- [x] Update CI for python 3.7 and tensorflow2.2.
- [x] Add doc for each model's call function.

### CODE REVIEW

**Comment by dathudeptrai** [comment]
@azraelkuan @l4zyf9x @MokkeMeguru PTAL before i merge it :)).
